packages/contracts-bedrock/src/L1/SuperchainConfig.sol

```diff
@@ -0,0 +1,260 @@
+    constructor() {
+        initialize({
+            _systemOwner: address(0),
+            _initiator: address(0),
+            _vetoer: address(0),
+            _guardian: address(0),
+            _delay: 0,
+            _maxPause: 0,
+            _sequencers: new Types.SequencerKeyPair[](0)
+        });
+    }
+
+    /// @notice Initializer.
+    ///         The resource config must be set before the require check.
+    /// @param _systemOwner Owner of the contract.
+    /// @param _initiator   Address of the initiator who may initiate an upgrade or change to critical config values.
+    /// @param _vetoer      Address of the vetoer.
+    /// @param _guardian    Address of the guardian, can pause the OptimismPortal.
+    /// @param _delay       The delay time in seconds between when an upgrade is initiated and when it can be finalized.
+    /// @param _maxPause    The maximum time in seconds that the system can be paused for.
+    /// @param _sequencers  The initial list of allowed sequencers
+    function initialize(
+        address _systemOwner,
+        address _initiator,
+        address _vetoer,
+        address _guardian,
+        uint256 _delay,
+        uint256 _maxPause,
+        Types.SequencerKeyPair[] memory _sequencers
+    )
+        public
+        reinitializer(2)
+    {
+        _setSystemOwner(_systemOwner);
+        _setInitiator(_initiator);
+        _setVetoer(_vetoer);
+        _setGuardian(_guardian);
+        _setDelay(_delay);
+        _setMaxPause(_maxPause);
+
+        for (uint256 i = 0; i < _sequencers.length; i++) {
+            _addSequencer(_sequencers[i]);
+        }
+    }
+
+    /// @notice Getter for the systemOwner address.
+    function systemOwner() public view returns (address systemOwner_) {
+        systemOwner_ = Slot.getAddress(SYSTEM_OWNER_SLOT);
+    }
+
+    /// @notice Getter for the initiator address.
+    function initiator() public view returns (address initiator_) {
+        initiator_ = Slot.getAddress(INITIATOR_SLOT);
+    }
+
+    /// @notice Getter for the vetoer address.
+    function vetoer() public view returns (address vetoer_) {
+        vetoer_ = Slot.getAddress(VETOER_SLOT);
+    }
+
+    /// @notice Getter for the guardian address.
+    function guardian() public view returns (address guardian_) {
+        guardian_ = Slot.getAddress(GUARDIAN_SLOT);
+    }
+
+    /// @notice Getter for the delay time.
+    function delay() public view returns (uint256 delay_) {
+        delay_ = Slot.getUint(DELAY_SLOT);
+    }
+
+    /// @notice Getter for the maxPause time.
+    function maxPause() public view returns (uint256 maxPause_) {
+        maxPause_ = Slot.getUint(MAX_PAUSE_SLOT);
+    }
+
+    /// @notice Getter for the paused address.
+    function paused() public view returns (bool paused_) {
+        paused_ = Slot.getUint(PAUSED_SLOT) > block.timestamp;
+    }
+
+    // todo(maurelian): we might need a repause() getter which is only callable by the
```

Comment:
Following up on this `pause`/`repause` - what is the thinking here?

Reply:
We do need this, but I would prefer to include it in a follow up PR just to get this closed and so that that can happen in the context of a smaller diff.